sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_request_tracing_context.py

```diff
@@ -0,0 +1,326 @@
+# license information.
+# -------------------------------------------------------------------------
+import os
+from typing import Dict, Optional, List, Tuple
+from importlib.metadata import version, PackageNotFoundError
+from ._constants import (
+    REQUEST_TRACING_DISABLED_ENVIRONMENT_VARIABLE,
+    ServiceFabricEnvironmentVariable,
+    AzureFunctionEnvironmentVariable,
+    AzureWebAppEnvironmentVariable,
+    ContainerAppEnvironmentVariable,
+    KubernetesEnvironmentVariable,
+    CUSTOM_FILTER_KEY,
+    PERCENTAGE_FILTER_KEY,
+    TIME_WINDOW_FILTER_KEY,
+    TARGETING_FILTER_KEY,
+    PERCENTAGE_FILTER_NAMES,
+    TIME_WINDOW_FILTER_NAMES,
+    TARGETING_FILTER_NAMES,
+    FEATURE_FLAG_USES_SEED_TAG,
+    FEATURE_FLAG_USES_TELEMETRY_TAG,
+)
+
+
+class HostType:
+    UNIDENTIFIED = ""
+    AZURE_WEB_APP = "AzureWebApp"
+    AZURE_FUNCTION = "AzureFunction"
+    CONTAINER_APP = "ContainerApp"
+    KUBERNETES = "Kubernetes"
+    SERVICE_FABRIC = "ServiceFabric"
+
+
+class RequestType:
+    STARTUP = "Startup"
+    WATCH = "Watch"
+
+
+Delimiter = "+"
+
+
+class _RequestTracingContext:  # pylint: disable=too-many-instance-attributes
+    """
+    Encapsulates request tracing and telemetry configuration values.
+    """
+
+    def __init__(self, load_balancing_enabled: bool = False) -> None:
+        # Main feature tracking properties
+        self.uses_load_balancing = load_balancing_enabled
+        self.uses_ai_configuration = False
+        self.uses_aicc_configuration = False  # AI Chat Completion
+        self.uses_telemetry = False
+        self.uses_seed = False
+        self.max_variants: Optional[int] = None
+        self.feature_filter_usage: Dict[str, bool] = {}
+        self.is_key_vault_configured: bool = False
+        self.replica_count: int = 0
+        self.is_failover_request: bool = False
+
+        # Host and environment detection properties
+        self.host_type: str = _RequestTracingContext.get_host_type()
+
+        # Version tracking
+        self.feature_management_version: Optional[str] = None
+
+    def update_max_variants(self, size: int) -> None:
+        """
+        Update max_variants if the new size is larger.
+
+        :param size: size of variant
+        :type size: int
+        """
+        if self.max_variants is None or size > self.max_variants:
+            self.max_variants = size
+
+    def _get_features_string(self) -> str:
+        """
+        Generate the features string for correlation context.
+
+        :return: A string containing features used separated by delimiters.
+        :rtype: str
+        """
+        features_list = []
+
+        if self.uses_load_balancing:
+            features_list.append("LB")
+        if self.uses_ai_configuration:
+            features_list.append("AI")
+        if self.uses_aicc_configuration:
+            features_list.append("AICC")
+        if self.get_assembly_version("azure-ai-projects"):
+            features_list.append("USE_AI")
```

Comment:
Change to `USE_AI_FOUNDRY_SDK`

Reply:
nit: How about `USES_AI_FOUNDRY_SDK` or `HAS_AI_FOUNDRY_SDK`?

Reply:
Also, could the version be useful to us?